python-package/lightgbm/dask.py

```diff
@@ -169,7 +181,16 @@ def _split_to_parts(data, is_matrix):
     return parts
 
 
-def _train(client, data, label, params, model_factory, sample_weight=None, group=None, **kwargs):
+def _train(
+    client: Client,
+    data: _DaskCollection,
+    label: _DaskCollection,
+    params: Dict[str, Any],
+    model_factory: Type[LGBMModel],
+    sample_weight: Optional[_DaskCollection] = None,
+    group: Optional[_1DArrayLike] = None,
```

Comment:
I'll update the docs in this PR too, I think that's directly related to this scope. Thanks!

Reply:
Thinking through this, I realized that with the way we do doc inheritance, it will actually be a bit tricky to override the docs for `group`.

I wrote up my thoughts in https://github.com/microsoft/LightGBM/issues/3871, but I won't update the docs (except those on internal functions) in this PR

Reply:
fixed the hint in https://docs.dask.org/en/latest/dataframe-api.html#dask.dataframe.to_csv